packages/cli/src/commands/upgrade/upgrade.js

```diff
@@ -60,9 +57,32 @@ const getPatch = async (currentVersion, newVersion, projectDir) => {
     return null;
   }
 
-  return patch
-    .replace(new RegExp(rnDiffAppName, 'g'), name)
-    .replace(new RegExp(rnDiffAppName.toLowerCase(), 'g'), name.toLowerCase());
+  let patchWithRenamedProjects = patch;
+
+  // rn-diff-purge only supports iOS and Android platforms
+  if (config.project.ios) {
+    patchWithRenamedProjects = patchWithRenamedProjects.replace(
+      new RegExp('RnDiffApp', 'g'),
+      // $FlowFixMe - poor typings of ProjectConfigIOST
+      config.project.ios.projectName.replace('.xcodeproj', ''),
+    );
+  }
+
+  if (config.project.android) {
+    patchWithRenamedProjects = patchWithRenamedProjects
+      .replace(
+        new RegExp('com\\.rndiffapp', 'g'),
+        // $FlowFixMe - poor typings of ProjectConfigAndroidT
```

Comment:
Maybe you would add the typing for these? At lest for the ones you are using, we can extend the types partially.

Reply:
I tried, but it seemed to result in quite a bunch of errors I didn't want to handle in this diff. Will try to address that later